### PR TITLE
Tlspe-20 TLSPE-21 TLSPE-22

### PR DESCRIPTION
**Pull request by dlhaines**
Add ability to manage grade input/output with files as well as ESB.  Rename some things for clarity.

### CODE REVIEW

File: src/main/java/edu/umich/its/spe/FileIO.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+		File file = new File(fileName);
+		String gradesString;
+
+		try {
+			// allow input file to have line breaks.
+			gradesString = FileUtils.readFileToString(file,"UTF-8").replaceAll("\\r|\\n", " ");
+		} catch (IOException e) {
+			String msg = "FileIO: exception: "+e.getMessage() + " file: "+fileName;
+			M_log.info(msg);
```
Comment (pushyamig) [inline src/main/java/edu/umich/its/spe/FileIO.java]: Why is this a info message?

File: src/main/java/edu/umich/its/spe/FileIO.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	public static WAPIResultWrapper getGradesFileStatic(SPEProperties speproperties, String gradedAfterTime) throws GradeIOException {
+		M_log.debug("getGrades FileIO: properties: {} gradedAfterTime: {}",speproperties,gradedAfterTime);
+		if (speproperties == null || gradedAfterTime == null || gradedAfterTime.length() == 0) {
+			throw new GradeIOException("FileIO Call invalid: speproperties: "+ speproperties+" gradedAfterTime: "+gradedAfterTime);
+		}
+
+		HashMap<String,String> ioProperties = speproperties.getIo();
+		String fileName = SPEUtils.safeGetPropertyValue(ioProperties,"getGradeIO");
+
+		File file = new File(fileName);
```
Comment (pushyamig) [inline src/main/java/edu/umich/its/spe/FileIO.java]: The above 3 lines are repeated below in the `putGradeFileStatic`can take this out as small method and pass in get/putGradeIO as argument

Comment (dlhaines) [reply src/main/java/edu/umich/its/spe/FileIO.java]: Good idea.

File: src/main/java/edu/umich/its/spe/FileIO.java (hunk 1 above)
Comment (dlhaines) [reply src/main/java/edu/umich/its/spe/FileIO.java]: They are promoted to error level now.

**Pull request merged**